include/boost/geometry/strategies/geographic/buffer_point_circle.hpp

```diff
@@ -0,0 +1,127 @@
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_BUFFER_POINT_CIRCLE_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_BUFFER_POINT_CIRCLE_HPP
+
+#include <cstddef>
+
+#include <boost/range.hpp>
+
+#include <boost/geometry/util/math.hpp>
+
+#include <boost/geometry/strategies/buffer.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace buffer
+{
+
+/*!
+\brief Create a circular buffer around a point, on the Earth
+\ingroup strategies
+\details This strategy can be used as PointStrategy for the buffer algorithm.
+    It creates a circular buffer around a point, on the Earth. It can be applied
+    for points and multi_points.
+
+\qbk{
+[heading Example]
+[buffer_geographic_point_circle]
+[buffer_geographic_point_circle_output]
+[heading See also]
+\* [link geometry.reference.algorithms.buffer.buffer_7_with_strategies buffer (with strategies)]
+\* [link geometry.reference.strategies.strategy_buffer_point_circle point_circle]
+\* [link geometry.reference.strategies.strategy_buffer_point_square point_square]
+}
+ */
+template
+<
+    typename FormulaPolicy = strategy::andoyer,
+    typename Spheroid = srs::spheroid<double>,
+    typename CalculationType = void
+>
+class geographic_point_circle
+{
+public :
+    //! \brief Constructs the strategy
+    //! \param count number of points for the created circle (if count
+    //! is smaller than 3, count is internally set to 3)
+    explicit geographic_point_circle(std::size_t count = 90)
+        : m_count((count < 3u) ? 3u : count)
+    {}
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+    //! Fills output_range with a circle around point using distance_strategy
+    template
+    <
+        typename Point,
+        typename OutputRange,
+        typename DistanceStrategy
+    >
+    inline void apply(Point const& point,
+                DistanceStrategy const& distance_strategy,
+                OutputRange& output_range) const
+    {
+        typedef typename boost::range_value<OutputRange>::type output_point_type;
+
+        typedef typename select_most_precise
+            <
+                typename geometry::coordinate_type<Point>::type,
+                typename geometry::coordinate_type<output_point_type>::type,
+                CalculationType
+                //double
+            >::type calculation_type;
+
+        calculation_type const buffer_distance = distance_strategy.apply(point, point,
+                        strategy::buffer::buffer_side_left);
+
+        typedef typename FormulaPolicy::template direct
+            <
+                calculation_type, true, false, false, false
+            > direct_t;
+
+        calculation_type const two_pi = geometry::math::two_pi<calculation_type>();
+
+        calculation_type const diff = two_pi / calculation_type(m_count);
+        // TODO: after calculation of some angles is corrected,
+        // we can start at 0.0
+        calculation_type angle = 0.001;
```

Comment:
I wanted to fix it. How can I verify that the angles are correct or incorrect? I didn't find any test case in the test file which was commented out due to this. Do you have in mind this issue: https://github.com/boostorg/geometry/issues/542 or is this a different issue?

I'm asking because I wanted to fix it by making the azimuth passed into the direct formula to be in valid range (-pi, pi], like this in the loop below:

    if (angle > pi)
        angle -= two_pi;

and also to start from 0. I verified that this changes the coordinates but causes more cases to fail. And I don't know if this is because the coordinates are incorrect or that it's inaccurate area's fault.

My understanding is that this `0.001` magic number only hides the problem in the specific several cases in test file. Is that correct? Because in general case if the intermediate results from some formulas are incorrect it doesn't matter where is the starting point since problematic angles can be achieved in various ways (i.e. by also tweaking the point count or changing the starting point).